homeassistant/components/powerfox/sensor.py

```diff
@@ -0,0 +1,150 @@
+"""Sensors for Powerfox integration."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Generic, TypeVar
+
+from powerfox import Device, PowerMeter, WaterMeter
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.const import UnitOfEnergy, UnitOfPower, UnitOfVolume
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import StateType
+
+from . import PowerfoxConfigEntry
+from .coordinator import PowerfoxDataUpdateCoordinator
+from .entity import PowerfoxEntity
+
+T = TypeVar("T", PowerMeter, WaterMeter)
+
+
+@dataclass(frozen=True, kw_only=True)
+class PowerfoxSensorEntityDescription(Generic[T], SensorEntityDescription):
+    """Describes Poweropti sensor entity."""
+
+    value_fn: Callable[[T], StateType]
```

Comment:
Ideally we are more specific here than StateType (I know we used to use this a lot, but we've recently had some discussion about it

Reply:
So what exactly is the suggestion to change this? Or is it still an ongoing discussion?

Reply:
change `StateType` to something more specific like `float | None`